File: app/src/main/java/github/leavesc/filetransfer/callback/DirectActionListener.java
package github.leavesc.filetransfer.callback;

import android.net.wifi.p2p.WifiP2pDevice;
import android.net.wifi.p2p.WifiP2pInfo;
import android.net.wifi.p2p.WifiP2pManager;

import java.util.Collection;

/**
 * @Author: leavesC
 * @Date: 2019/2/27 23:58
 * @Desc:
 * @Github：https://github.com/leavesC
 */
public interface DirectActionListener extends WifiP2pManager.ChannelListener {

    void wifiP2pEnabled(boolean enabled);

    void onConnectionInfoAvailable(WifiP2pInfo wifiP2pInfo);

    void onDisconnection();

    void onSelfDeviceAvailable(WifiP2pDevice wifiP2pDevice);

    void onPeersAvailable(Collection<WifiP2pDevice> wifiP2pDeviceList);

}

File: app/src/main/java/github/leavesczy/wifip2p/service/DirectActionListener.kt
package github.leavesczy.wifip2p.service

import android.net.wifi.p2p.WifiP2pDevice
import android.net.wifi.p2p.WifiP2pInfo
import android.net.wifi.p2p.WifiP2pManager.ChannelListener

/**
 * @Author: leavesCZY
 * @Desc:
 */
interface DirectActionListener : ChannelListener {
    fun wifiP2pEnabled(enabled: Boolean)
    fun onConnectionInfoAvailable(wifiP2pInfo: WifiP2pInfo)
    fun onDisconnection()
    fun onSelfDeviceAvailable(wifiP2pDevice: WifiP2pDevice?)
    fun onPeersAvailable(wifiP2pDeviceList: Collection<WifiP2pDevice>)
}